tests/e2e/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,307 @@
+// +build e2e
+
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package standalone_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core_v1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/dapr/cli/pkg/kubernetes"
+	"github.com/dapr/cli/tests/e2e/spawn"
+
+	k8s "k8s.io/client-go/kubernetes"
+
+	//  azure auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
+
+	//  gcp auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	//  oidc auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	//  openstack auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/openstack"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	daprNamespace        = "dapr-cli-tests"
+	daprRuntimeVersion   = "1.0.1"
+	daprDashboardVersion = "0.6.0"
+)
+
+func TestKubernetesInstall(t *testing.T) {
+	// Ensure a clean environment
+	uninstall()
+
+	tests := []struct {
+		name  string
+		phase func(*testing.T)
+	}{
+		{"test install", testInstall},
+		{"check crds", testCRDs},
+		{"check clusterroles", testClusterRoles},
+		{"check clusterrolebindings", testClusterRoleBindings},
+		{"test uninstall", testUninstall},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, tc.phase)
+	}
+}
+
+func getDaprPath() string {
+	distDir := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+
+	return filepath.Join("..", "..", "..", "dist", distDir, "release", "dapr")
+}
+
+func uninstall() (string, error) {
+	daprPath := getDaprPath()
+
+	return spawn.Command(daprPath,
+		"uninstall", "-k",
+		"-n", daprNamespace,
+		"--log-as-json")
+}
+
+func testUninstall(t *testing.T) {
```

Comment:
should a check be added to test CRD and role removal also?

Reply:
I will do some magic to test reverse polarity.